### ISSUE TITLE
Inconsistent use of the # sign in error messages

### ISSUE THREAD

**Issue by Mertsch** [FakeItEasy/FakeItEasy#1278]
As you can see in the following example, the error message tries to be very natural in its language.
This should also be applied to the 2nd number in the text `#7`.

Have it be "never", "once", "n times" like the first number.

```
Assertion failed for the following call:
    SomeMethodName()
  Expected to find it exactly 6 times but found it #7 times among the calls:
[...]
```

**Comment by blairconrad** [comment]
Oh, thanks for bringing this up, @Mertsch. Is it something you're interested in fixing?

**Comment by Mertsch** [comment]
@blairconrad Yes.
Which variant would you choose?

`but found it #{0} times among the calls`
A
```
1: once
2: twice
n: n times
```

B
```
n: n times
```

**Comment by Mertsch** [comment]
I would change the code here and add a `switch` for case A or just remove the # sign for B
https://github.com/FakeItEasy/FakeItEasy/blob/master/src/FakeItEasy/Core/FakeAsserter.cs#L64

**Comment by blairconrad** [comment]
> Which variant would you choose?

Good question! I'd been thinking about it.  The switch is a little bit of work, but I think maybe 

```
Expected to find it at least twice but found it {once/twice/n times} among the calls
```

I mean, "1 times" would read funny anyhow, so if you're going to have to switch, why not match the language of our built-in convenience methods?

> I would change the code here and add a switch for case A or just remove the # sign for B
> https://github.com/FakeItEasy/FakeItEasy/blob/master/src/FakeItEasy/Core/FakeAsserter.cs#L64

Sounds fantastic. Note that a number of tests will break when you do this, but for all the right reasons.

**Comment by Mertsch** [comment]
@blairconrad PR is out, and I have added 4 new tests

![image](https://user-images.githubusercontent.com/9402861/33826528-9a999ea0-de65-11e7-9f4b-1597f8925ace.png)

**Comment by blairconrad** [comment]
Thanks, @Mertsch!

**Comment by blairconrad** [comment]
@thomaslevesque, I see you added this to the release notes as a "Changed".  Technically true, but I'd be inclined to classify as "Fixed" to match the "bug" we've called this issue, and so the SemVer police don't think we need a 5.0.0 release for this. Is that acceptable?

**Comment by thomaslevesque** [comment]
> @thomaslevesque, I see you added this to the release notes as a "Changed". Technically true, but I'd be inclined to classify as "Fixed" to match the "bug" we've called this issue, and so the SemVer police don't think we need a 5.0.0 release for this. Is that acceptable?

Ah, you're right. I didn't notice the `bug` label. Of course, it should be in "Fixed". I'll change it now.

**Comment by blairconrad** [comment]
Thanks, sir.

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 4.4.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/4.4.0).

Thanks for your help with this, @mertsch. Look for your name in the release notes! :trophy:
